src/bp-core/bp-core-functions.php

```diff
@@ -6060,4 +6060,86 @@ function bb_moderation_bg_update_moderation_data() {
 	if ( ! empty( $moderated_activities ) ) {
 		bb_moderation_update_suspend_data( $moderated_activities, 0 );
 	}
-}
\ No newline at end of file
+}
+
+/**
+ * Function will restrict RSS feed.
+ *
+ * @since BuddyBoss x.x.x
+ */
+function bb_restricate_rss_feed() {
+	// This will disable default feeds.
+	remove_action( 'do_feed_rdf', 'do_feed_rdf', 10, 1 );
+	remove_action( 'do_feed_rss', 'do_feed_rss', 10, 1 );
+	remove_action( 'do_feed_rss2', 'do_feed_rss2', 10, 1 );
+	remove_action( 'do_feed_atom', 'do_feed_atom', 10, 1 );
+	remove_action( 'wp_head', 'feed_links_extra', 3 );
+	remove_action( 'wp_head', 'feed_links', 2 );
+	// This will disable sitewide feeds.
+	remove_action( 'bp_actions', 'bp_activity_action_sitewide_feed' );
+	// This will disable users personal activity feeds.
+	remove_action( 'bp_actions', 'bp_activity_action_personal_feed' );
+	// This will disable users friends activity feeds.
+	remove_action( 'bp_actions', 'bp_activity_action_friends_feed' );
+	// This will disable users group activity feeds.
+	remove_action( 'bp_actions', 'bp_activity_action_my_groups_feed' );
+	// This will disable users mentions activity feeds.
+	remove_action( 'bp_actions', 'bp_activity_action_mentions_feed' );
+	// This will disable users favorites activity feeds.
+	remove_action( 'bp_actions', 'bp_activity_action_favorites_feed' );
+	// This will disable groups feeds.
+	remove_action( 'bp_actions', 'groups_action_group_feed' );
+	// This will disable forums feed.
+	remove_filter( 'bbp_request', 'bbp_request_feed_trap' );
+}
+
+/**
+ * Function will restrict REST API.
+ *
+ * @since BuddyBoss x.x.x
+ */
+function bb_restricate_rest_api() {
+	// This will disable woocommerce endpoints.
+	if ( class_exists( 'woocommerce' ) ) {
+		remove_action( 'rest_api_init', array( WC()->api, 'register_rest_routes' ), 10 );
+	}
```

Comment:
@jitendrabanjara1991 Remove this as we don't need to control or restrict other plugin API.